docs/toctoc/guides/token-based-recovery-flow.md

```diff
@@ -0,0 +1,180 @@
+---
+id: token-based-recovery-flow
+title: Token-based Recovery Flow
+---
+
+This flow allows to implement the password recovery functionality of a web
+application.
+
+## Authentication domains
+
+The flow defines and orchestrates two
+[authentication domains](toctoc/concepts/functional-model.md):
+
+- **`LoginDomain`**: an authentication domain whose credential is the `Login`
+  data type.
+- **`AccessTokenDomain`**: an authentication domain whose credential is the
+  `AccessToken` data type.
+
+The `Login` data type is defined by a username and a password, while the
+`AccessToken` data type is defined by a value -- which is usually a randomly
+generated string -- and an expiration date.
+
+> 👉 Note: both the token-based authentication flow and the token-based recovery
+> flow use the same kind of domain (`AccessTokenDomain`).
+>
+> However, you should normally have **two separate instances** for those
+> domains, as their semantics and policies are likely to be different.
+>
+> For example, you usually want to limit the password recovery access token to
+> the recovery functionality, and use an auth token for anything else.
+
+## Functionalities
+
+The flow is implemented by the `TokenBasedRecoveryFlow` class, which exposes
+these functionalities:
+
+- `registerSubjectLogin`: registers a `Subject` for recovery, by creating an
```

Comment:
Minor comment: here I was a little confused because there is a method called `registerSubjectLogin` also in the token based auth flow, but it does a different thing: it signs up a new user given a password, which is similar to what here is done by the `recoverLogin` method. Maybe this should be called `exchangeForTokens` or similar?

Reply:
Oh wait, reading the code this seems just a typo because I see `registerForRecovery`

Reply:
right, it's a typo, sorry.